docs/src/test/scala/docs/home/state/BlogPostJsonColumn.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2022 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package docs.home.state
+
+import akka.persistence.r2dbc.state.scaladsl.AdditionalColumn
+// #additional-column-json
+import io.r2dbc.postgresql.codec.Json
+
+class BlogPostJsonColumn extends AdditionalColumn[BlogPost.State, Json] {
+
+  override val columnName: String = "query_json"
+
+  override def bind(upsert: AdditionalColumn.Upsert[BlogPost.State]): AdditionalColumn.Binding[Json] =
+    upsert.value match {
+      case s: BlogPost.DraftState =>
+        val jsonString = s"""{"title": "${s.content.title}", "published": false}"""
+        val json = Json.of(jsonString)
+        AdditionalColumn.BindValue(json)
+      case s: BlogPost.PublishedState =>
+        val jsonString = s"""{"title": "${s.content.title}", "published": true}"""
```

Comment:
Fine like this for a sample but note that it is not quite safe as quotes in the title string would break it (vs some proper JSON serialization solution).

Reply:
yeah, let me add a comment that a json library would be used